libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -343,6 +361,14 @@ void ContractCompiler::appendFunctionSelector(ContractDefinition const& _contrac
 	}
 
 	FunctionDefinition const* fallback = _contract.fallbackFunction();
+
+	solAssert(!_contract.isLibrary() || !fallback, "Libraries can't have fallback functions");
+
+	bool noPayableFunctions = !hasPayableFunctions(fallback, interfaceFunctions);
```

Comment:
```suggestion
	bool hasPayableFunctions = hasPayableFunctions(fallback, interfaceFunctions);
```
This would prevent to "double-negation" further down: ``!noPayableFunctions``